src/create-model.jl

```diff
@@ -1,37 +1,114 @@
-export create_model!, create_model, construct_dataframes, filter_assets, filter_flows
+export create_model!, create_model, construct_dataframes
 
 """
-    Helper function to filter assets in the graph given a key and value
+    _get_graph_asset_or_flow(graph, a)
+    _get_graph_asset_or_flow(graph, (u, v))
+
+Returns `graph[a]` or `graph[u, v]`.
+"""
+_get_graph_asset_or_flow(graph, a) = graph[a]
+_get_graph_asset_or_flow(graph, f::Tuple) = graph[f...]
+
 """
-filter_assets(graph, assets, key, values) =
-    filter(a -> !ismissing(getfield(graph[a], key)) && getfield(graph[a], key) == values, assets)
-filter_assets(graph, assets, key, values::Vector{String}) =
-    filter(a -> !ismissing(getfield(graph[a], key)) && getfield(graph[a], key) in values, assets)
+    get_graph_value_or_missing(graph, graph_key, field_key)
+    get_graph_value_or_missing(graph, graph_key, field_key, year)
+
+Get `graph[graph_key].field_key` (or `graph[graph_key].field_key[year]`) or return `missing` if
+any of the values do not exist.
+We also check if `graph[graph_key].active[year]` is true if the `year` is passed and return
+`missing` otherwise.
+"""
+function get_graph_value_or_missing(graph, graph_key, field_key)
+    g = _get_graph_asset_or_flow(graph, graph_key)
+    # This is not missing in the tests
+    # if g === missing
+    #     return missing
+    # end
+    # if !hasproperty(g, field_key)
+    #     return missing
+    # end
```

Comment:
should we leave this?

Reply:
The best solution would be to have these cases in the tests. But then how to make that happen exactly, I'm not sure.
If we enable the check, then the coverage decreases.
Removing would be easier. Do you agree with that?